source/k64f_eth_nanostack_port.c

```diff
@@ -17,83 +17,101 @@
 
 
 #include <string.h>
-#include <stdbool.h>
-#include "platform/arm_hal_phy.h"
-#include "platform/arm_hal_interrupt.h"
+#include <stdlib.h>
+#include <stdio.h>
+#include "arm_hal_phy.h"
+#include "arm_hal_interrupt.h"
 #include "ns_types.h"
 #include "k64f_eth_nanostack_port.h"
-#include "phy_link_status_wrapper.h"
-#include "fsl_enet_driver.h"
-#include "fsl_enet_hal.h"
-#include "fsl_phy_driver.h"
-#include "mbed-drivers/mbed_interface.h"
-#include "fsl_interrupt_manager.h"
+#include "fsl_enet.h"
+#include "fsl_phy.h"
+#include "eventOS_event_timer.h"
 #define HAVE_DEBUG 1
 #include "ns_trace.h"
-#include "ns_list.h"
 #include "nsdynmemLIB.h"
+#include "cmsis_os.h"
 
 
-/* Any Pre-processor Macros*/
-#define AUTO_INDEX (-1)
+/* Macro Definitions */
 #ifndef MEM_ALLOC
 #define MEM_ALLOC ns_dyn_mem_alloc
 #endif
 #ifndef MEM_FREE
 #define MEM_FREE ns_dyn_mem_free
 #endif
-#define TRACE_GROUP  "ethDrv"
+#define TRACE_GROUP  "Eth"
+#define ENET_HDR_LEN                  (14)
+#define ENET_RX_RING_LEN              (8)
+#define ENET_TX_RING_LEN              (4)
+#define TX_BUF_ALIGNMENT              (8U)
+#define ENET_ETH_MAX_FLEN             (1518)
+#define ENET_BuffSizeAlign(n) ENET_ALIGN(n, ENET_BUFF_ALIGNMENT)
```

Comment:
You never use this define. And I see a slight mixture of TX_BUFF_ALIGNMENT/RX_BUFF_ALIGNMENT/BUFF_ALIGNMENT. May be worth double checking the various alignment calls.